### PR TITLE
Added AutoCloseableLock to ensure the lock unlocks every time

### PR DESCRIPTION
**Pull request by AsafMah**
Added AutoCloseableLock to ensure the lock unlocks every time.

Solves problem of deadlock when exception is thrown.

### CODE REVIEW

**Review by yihezkel — APPROVED** [review]

File: src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java [hunk 1]
```diff
@@ -205,14 +204,11 @@ void writeRecord(SinkRecord record) throws ConnectException {
       if (record == null) {
         this.currentOffset = record.kafkaOffset();
       } else {
-        try {
-          reentrantReadWriteLock.readLock().lock();
+        try (AutoCloseableLock ignored = new AutoCloseableLock(reentrantReadWriteLock.readLock())) {
           this.currentOffset = record.kafkaOffset();
           fileWriter.writeData(record);
-        } catch (IOException | DataException ex) {
+        } catch (Exception ex) {
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:210]: revert

[the remark this answers is not in the document]
Comment (AsafMah) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:210]: image
Using try with-resources requires handling all exceptions

[the remark this answers is not in the document]
Comment (AsafMah) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java:210]: image
Using try with-resources requires handling all exceptions

[the remark this answers is not in the document]
Comment (AsafMah) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java]: Fixed.

[the remark this answers is not in the document]
Comment (AsafMah) [reply src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java]: Fixed

**Review by ohadbitt — APPROVED** [review]

**Pull request merged**